### DISCUSSION TITLE
How to get terms required for DDP

### DISCUSSION THREAD

**Discussion by lwygzh** [stack-of-tasks/pinocchio#2057] [General]
Hi,

Thank you for open sourcing this library!

I am new to this library, I was wondering after creating the robot model using

```python
robot = pinocchio.robot_wrapper.RobotWrapper.BuildFromURDF(urdf_path)
```

could I get the functions $x_{n+1} = f(x_n, u_n)$, $\displaystyle\frac{\partial f}{\partial u}$, $\displaystyle\frac{\partial f}{\partial x}$, $\displaystyle\frac{\partial^2 f}{\partial x\partial x}$, $\displaystyle\frac{\partial^2 f}{\partial x\partial u}$ and $\displaystyle\frac{\partial^2 f}{\partial u\partial u}$ using pinocchio?

**Comment by stephane-caron** [comment]
The answer to your question, where $f$ is the forward dynamics function of your system, is yes. You can look for instance at [Crocoddyl](https://github.com/loco-3d/crocoddyl), which implements DDP using Pinocchio. You can also look at the [`forward-dynamics-derivatives.cpp`](https://github.com/stack-of-tasks/pinocchio/blob/4512214cf08e765a28542c75bf6378e934622a38/examples/forward-dynamics-derivatives.cpp) or [`forward-dynamics-derivatives.py`](https://github.com/stack-of-tasks/pinocchio/blob/4512214cf08e765a28542c75bf6378e934622a38/examples/forward-dynamics-derivatives.py) examples.

**Reply by lwygzh** [reply]
Thanks for the response!
